Framework/Core/include/Framework/AnalysisHelpers.h

```diff
@@ -84,13 +134,13 @@ struct WritingCursor<soa::Table<ORIGIN, PC...>> {
   decltype(FFL(std::declval<cursor_t>())) cursor;
 
  private:
-  template <typename T>
-  static decltype(auto) extract(T const& arg)
+  template <typename A>
+  static decltype(auto) extract(A const& arg)
   {
-    if constexpr (soa::is_soa_iterator_v<T>) {
+    if constexpr (soa::is_iterator<A>) {
       return arg.globalIndex();
     } else {
-      static_assert(!framework::has_type<T>(framework::pack<PC...>{}), "Argument type mismatch");
+      static_assert(!framework::has_type<A>(typename persistent_table_t::persistent_columns_t{}), "Argument type mismatch");
```

Comment:
I would use a `requires arg.globalIndex()` and drop the static_assert.

Reply:
Implemented in #13679